docs/api/features/placement.md

```diff
@@ -1,3 +1,137 @@
+    / # hostname
+    98d5a22a0c5e
+    / # /native/usr/sbin/mdata-get sdc:server_uuid
+    44454c4c-4400-1054-8052-b5c04f383432
+
+Outside the containers, the physical server on which a container is running
+is exposed via the `compute_node` field using the [Triton CLI](XXX):
+
+    $ triton insts -o shortid,name,age,compute_node
+    SHORTID   NAME             AGE  COMPUTE_NODE
+    cc72c36c  serene_bose      6m   44454c4c-5400-1034-8052-b5c04f383432
+    98d5a22a  goofy_engelbart  2m   44454c4c-4400-1054-8052-b5c04f383432
+
+
+## Swarm affinity
+
+[Docker Swarm](https://docs.docker.com/swarm/overview/) is a system to allow you
+to talk to a pool of Docker hosts (a.k.a. nodes) as a single virtual Docker
+host. `docker run` against a Swarm master will choose a node on which to run
+your container. Sometimes it matters which node is selected -- my
+webhead should run near the redis it uses for caching, this database instance
+should *not* be on the same node as other database instances in the HA cluster.
+Swarm defines "filters" for node selection, and in particular ["affinity
+filters"](https://docs.docker.com/swarm/scheduler/filter/#use-an-affinity-filter)
+where node selection is described in terms of existing containers. For example:
+
+    docker run --name db1 -e 'affinity:container!=db0' -d mysql
+    #                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+    # Run a mysql instance that is NOT on the same node as container 'db0'.
+
+For the same reasons, node selection can matter on Triton. Triton's Docker
+implements the same affinity filters as documented for Docker Swarm, with the
+difference that you don't need to setup a Swarm cluster.
+
+
+### Affinity filter syntax
+
+An affinity filter is an argument to `docker run` or `docker create` using
+either (a) environment variables:
+
+    docker run -e 'affinity:<filter>' ...
+
+or (b) labels. (Note: At the time of writing the label syntax was in Docker
+Swarm [code](https://github.com/docker/swarm/blob/d9beef7/cluster/config.go#L83-L86),
+but not [documented](https://docs.docker.com/swarm/scheduler/filter/).)
+
+    docker run --label 'com.docker.swarm.affinities=["<filter>",...]' ...
+
+
+A `<filter>` is one of the following. (Note: Swarm defines "image filters"
+which don't apply for Triton because any image in the datacenter is available
+for any node.)
+
+- A container filter: `container<op><value>`
+- A label filter: `<label><op><value>`
+
+`<op>` is one of:
+
+- `==`: The new container must be on the same node as the container(s)
+  identified by `<value>`.
+- `!=`: The new container must be on a different node as the container(s)
+  identified by `<value>`.
+- `==~`: The new container should be on the same node as the container(s)
+  identified by `<value>`. I.e. this is a best effort or "soft" rule.
+- `!=~`: The new container should be on a different node as the container(s)
+  identified by `<value>`. I.e. this is a best effort or "soft" rule.
```

Comment:
Thanks, yes I'll add a note about this.

> E.g. when there is "container!=foo" and "container==~bar" in the request, the requirements are both evaluated as "hard".

Actually given both hard and soft affinities *the soft ones are ignored*: https://github.com/joyent/sdc-docker/blob/f1684e3f32335055986f1e47381d3494d070c506/lib/backends/sdc/affinity.js#L672-L677